persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -206,6 +215,17 @@ public void runWithinTransaction(TransactionCallback callback) throws SQLExcepti
         });
   }
 
+  public Integer execute(Connection connection, QueryGenerator.PreparedQuery preparedQuery)
+      throws SQLException {
+    try (PreparedStatement statement = connection.prepareStatement(preparedQuery.sql())) {
```

Comment:
This code appears to be called one per entity in `writeEntities`. I believe we should try and refactor the code to do `prepareStatement` one, execute many in that case.

Reply:
I did not mean preparing at startup, but at least reusing the `PreparedStatement` inside loops.